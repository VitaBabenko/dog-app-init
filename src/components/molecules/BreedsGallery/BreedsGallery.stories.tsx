import type { Meta, StoryObj } from '@storybook/react';
import { BreedsGallery } from '.';

const meta = {
  title: 'Molecules/BreedsGallery',
  component: BreedsGallery
} satisfies Meta<typeof BreedsGallery>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {
    page: 1
  }
};
