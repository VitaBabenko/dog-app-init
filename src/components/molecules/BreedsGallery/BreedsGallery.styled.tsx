import { Box, styled } from '@mui/material';

export const StyledBox = styled(Box)(({ theme }) => ({
  display: 'grid',
  gridTemplateColumns: 'repeat(3, 372px)',
  gridGap: 22,
  marginBottom: '25px',
  '& img': {
    marginBottom: '14px',
    height: '274px'
  },
  '& p': {
    padding: '0px 28px'
  },
  '& .MuiButton-text': {
    position: 'absolute',
    bottom: '8px',
    right: '5px',
    display: 'block',
    backgroundColor: 'transparent',
    color: theme.palette.warning.main
  }
}));
