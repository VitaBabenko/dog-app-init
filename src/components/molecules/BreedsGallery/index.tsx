import { Skeleton } from '@mui/material';
import { FC } from 'react';
import { BreedCardGallery } from '../../atoms/BreedCardGallery';
import { useGetBreedsQuery } from '../../../services/breeds';
import { StyledBox } from './BreedsGallery.styled';

type BreedsGalleryProps = {
  page: number;
};

export const BreedsGallery: FC<BreedsGalleryProps> = ({ page }) => {
  const {
    data: breeds,
    isLoading,
    isFetching
  } = useGetBreedsQuery({ limit: 6, page });

  console.log('isLoading', isLoading);
  console.log('isFetching', isFetching);

  return (
    <StyledBox>
      {isLoading && isFetching && (
        <Skeleton variant="rounded" width={1160} height={1000} />
      )}
      {(breeds || []).map(({ id, image, name, temperament }) => (
        <BreedCardGallery
          key={id}
          id={id}
          image={image}
          name={name}
          temperament={temperament}
        />
      ))}
    </StyledBox>
  );
};
