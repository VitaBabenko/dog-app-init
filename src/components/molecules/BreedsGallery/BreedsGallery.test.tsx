import { composeStories } from '@storybook/react';
import * as stories from './BreedsGallery.stories';
import { testSnapshot } from '../../../test/utils/snapshot.utils';

describe('BreedsGallery', () => {
  const { Primary } = composeStories(stories);
  testSnapshot(<Primary />);
});
