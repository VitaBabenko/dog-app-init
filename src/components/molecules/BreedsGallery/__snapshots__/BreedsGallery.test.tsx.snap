// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreedsGallery should match snapshot Component 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
  }

  50% {
    opacity: 0.4;
  }

  100% {
    opacity: 1;
  }
}

.emotion-0 {
  display: grid;
  grid-template-columns: repeat(3, 372px);
  grid-gap: 22px;
  margin-bottom: 25px;
}

.emotion-0 img {
  margin-bottom: 14px;
  height: 274px;
}

.emotion-0 p {
  padding: 0px 28px;
}

.emotion-0 .MuiButton-text {
  position: absolute;
  bottom: 8px;
  right: 5px;
  display: block;
  background-color: transparent;
  color: #FFCF32;
}

.emotion-1 {
  display: block;
  background-color: rgba(95, 85, 109, 0.11);
  height: 1.2em;
  border-radius: 4px;
  -webkit-animation: animation-0 2s ease-in-out 0.5s infinite;
  animation: animation-0 2s ease-in-out 0.5s infinite;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <span
      class="MuiSkeleton-root MuiSkeleton-rounded MuiSkeleton-pulse emotion-1"
      style="width: 1160px; height: 1000px;"
    />
  </div>
</div>
`;
